src/main/java/com/amazon/pvar/tspoc/merlin/solver/CallGraph.java

```diff
@@ -163,4 +182,22 @@ public String toString() {
                 .strip();
     }
 
+    private static String sourceLocationToJSON(SourceLocation src) {
+        return """
+            {
+              "source": "%s",
+              "start": {
+                "line": %d,
+                "column": %d
+              },
+              "end": {
+                "line": %d,
+                "column": %d
+              }
+            }    
+        """.formatted(src.getLocation().getFile().toString(), 
```

Comment:
This could cause issues with escaping if the file name contains characters like quotes or backspaces. Should we use something like gson to produce the json instead?